library/core/src/char/methods.rs

```diff
@@ -1118,7 +1118,7 @@ impl char {
     #[stable(feature = "ascii_methods_on_intrinsics", since = "1.23.0")]
     #[inline]
     pub fn to_ascii_lowercase(&self) -> char {
-        if self.is_ascii() { (*self as u8).to_ascii_lowercase() as char } else { *self }
+        if self.is_ascii_uppercase() { ((*self as u8) | ASCII_CASE_MASK) as char } else { *self }
```

Comment:
Would it be better to have a comment here explaining what is going on?

Reply:
We could do. There's a note on the const definition as to why it exists.

An alternative way to make it more readable would be to extract:
`((*self as u8) | ASCII_CASE_MASK) as char` as its own inlined method: `to_ascii_lowercase_unchecked()` and we could mark the method as unsafe to point out that callers should check that the input is ascii?

- I'm on the fence as to whether that would be better - it might be slightly nicer?

Reply:
I've made those changes. I think it's slightly nicer as rather than a pub(crate) constant we have a pub(crate) function to change the case and the constant is now used just in one file and near to each other. 

(bikeshedding: maybe `ascii_invert_case()` rather than `ascii_change_case()` ?)

Reply:
`toggle_ascii_case` might also work

Reply:
Reran the perf tests, still look the same improvement:
```
test char::methods::bench_to_ascii_lowercase                    ... bench:       8,473 ns/iter (+/- 1,276)
test char::methods::bench_to_ascii_uppercase                    ... bench:       8,484 ns/iter (+/- 2,071)
```
If people are happy with what we have I think we can merge this now.